plugins/mergeStyles.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const { querySelectorAll, closestByName } = require('../lib/xast.js');
+const { getCssStr, setCssStr } = require('../lib/css-tools');
+
+exports.type = 'full';
+exports.active = true;
+exports.description = 'merge multiple style elements into one';
+
+/**
+ * Merge multiple style elements into one.
+ *
+ * @param {Object} document document element
+ *
+ * @author strarsis <strarsis@gmail.com>
+ */
+exports.fn = function (document) {
+  // collect <style/>s with valid type attribute (preserve order)
+  const styleElements = querySelectorAll(document, 'style');
+
+  // no <styles/>s, nothing to do
+  if (styleElements.length <= 1) {
```

Comment:
Previous question probably hidden by github. This still prevents one empty style from removing.

Reply:
Indeed. I added a test and a fix for this.